File: ios-mma-recomp/MMARecomp/Models/Meal.swift
import Foundation

struct Meal: Codable, Identifiable {
    var id: UUID
    var userId: UUID
    var date: String
    var repas: Int
    var calories: Int
    var proteinesG: Double
    var glucidesG: Double
    var lipidesG: Double
    var description: String?
    var createdAt: String?

    var repasSlot: RepasSlot? { RepasSlot(rawValue: repas) }

    enum CodingKeys: String, CodingKey {
        case id
        case userId = "user_id"
        case date, repas, calories
        case proteinesG = "proteines_g"
        case glucidesG = "glucides_g"
        case lipidesG = "lipides_g"
        case description
        case createdAt = "created_at"
    }
}

struct NewMeal: Codable {
    var date: String
    var repas: Int
    var calories: Int
    var proteinesG: Double
    var glucidesG: Double
    var lipidesG: Double
    var description: String?

    enum CodingKeys: String, CodingKey {
        case date, repas, calories
        case proteinesG = "proteines_g"
        case glucidesG = "glucides_g"
        case lipidesG = "lipides_g"
        case description
    }
}

File: android-mma-recomp/app/src/main/java/com/example/mmarecomp/model/Meal.kt
package com.example.mmarecomp.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Meal(
    val id: String,
    @SerialName("user_id") val userId: String,
    val date: String,
    val repas: Int,
    val calories: Int,
    @SerialName("proteines_g") val proteinesG: Double,
    @SerialName("glucides_g") val glucidesG: Double,
    @SerialName("lipides_g") val lipidesG: Double,
    val description: String? = null,
    @SerialName("created_at") val createdAt: String? = null,
) {
    val repasSlot: RepasSlot? get() = RepasSlot.fromValue(repas)
}

@Serializable
data class NewMeal(
    val date: String,
    val repas: Int,
    val calories: Int,
    @SerialName("proteines_g") val proteinesG: Double,
    @SerialName("glucides_g") val glucidesG: Double,
    @SerialName("lipides_g") val lipidesG: Double,
    val description: String? = null,
)
